plugins/woocommerce/includes/class-woocommerce.php

```diff
@@ -871,16 +871,21 @@ public function init() {
 	 *      - WP_LANG_DIR/plugins/woocommerce-LOCALE.mo
 	 */
 	public function load_plugin_textdomain() {
-		$locale = determine_locale();
-
 		/**
 		 * Filter to adjust the WooCommerce locale to use for translations.
 		 */
-		$locale = apply_filters( 'plugin_locale', $locale, 'woocommerce' ); // phpcs:ignore WooCommerce.Commenting.CommentHooks.MissingSinceComment
-
-		unload_textdomain( 'woocommerce', true );
-		load_textdomain( 'woocommerce', WP_LANG_DIR . '/woocommerce/woocommerce-' . $locale . '.mo' );
-		load_plugin_textdomain( 'woocommerce', false, plugin_basename( dirname( WC_PLUGIN_FILE ) ) . '/i18n/languages' );
+		$locale                  = apply_filters( 'plugin_locale', determine_locale(), 'woocommerce' ); // phpcs:ignore WooCommerce.Commenting.CommentHooks.MissingSinceComment
+		$custom_translation_path = WP_LANG_DIR . '/woocommerce/woocommerce-' . $locale . '.mo';
+		$plugin_translation_path = WP_LANG_DIR . '/plugins/woocommerce-' . $locale . '.mo';
+
+		// If a custom translation exists (by default it will not, as it is not a standard WordPress convention)
+		// we unload the existing translation, then essentially layer the custom translation on top of the canonical
+		// translation. Otherwise, we simply step back and let WP manage things.
+		if ( is_readable( $custom_translation_path ) ) {
+			unload_textdomain( 'woocommerce' );
```

Comment:
I really like that we're now enabling the custom code only when a custom translation is already in place, but I wonder if we should remove the `$reloadable = true` arg in the `unload_textdomain()`. Depending on what other things happen during script execution, wouldn't this be useful to have? Have you seen that break something else?

Reply:
Right! One might think so. In practice, at least in my testing, this would mean that the "custom" translation (the one that translates *products* as *les produits)* will no longer take effect.

Reply:
To simulate, follow the testing instructions then modify this code as you suggest. On refreshing the dash, you should see that *products* is translated as *produits* (ie, custom translation is skipped).

Reply:
I didn't see that happening when I tried and I just tried again, but I might be missing something. At least on my instance (WP 6.7.1) the file wasn't skipped.
I don't think this is particularly serious but would be good if we could both reproduce the same results.

Reply:
Scratch that. So... things are more complicated as usual and I'm the one who's wrong here. If I don't force an "early" translation, things do work (despite the `$reload = true`), but if I do, then it works as you described. Weird, but 🤷.

Reply:
Yeah, that's now beyond our control (or, is beyond our control as of WP 6.7). At least as I currently understand things.